contracts/core/StakeInterface.sol

```diff
@@ -0,0 +1,151 @@
+
+    /**
+     * @notice Withdraw the deposit of a validator. A deposit can only be
+     *         withdrawn after the holding period has passed after the
+     *         validator logged out.
+     *
+     * @param _validatorIndex The unique index of the validator that shall
+     *                        approve the transfer of the deposit to its
+     *                        associated withdrawal address.
+     *
+     * @return `true` if the transfer was approved successfully.
+     */
+    function withdraw(
+        uint256 _validatorIndex
+    )
+        external
+        returns (bool success_);
+
+    /**
+     * @notice Slashes a validator based on two votes that violate a slashing
+     *         condition.
+     *
+     * @param _firstVoteMessage The first of the two votes, RLP encoded.
+     * @param _secondVoteMessage The second of the two votes, RLP encoded.
+     *
+     * @return `true` if the validator has been slashed.
+     */
+    function slash(
+        bytes _firstVoteMessage,
+        bytes _secondVoteMessage
+    )
+        external
+        returns (bool slashed_);
+
+    /**
+     * @notice Notifies the contract about a closing OSTblock in order to
+     *         handle any changes in the set of validators.
+     *         Can only be called from OriginCore.
+     *
+     * @param _closedOstBlockHeaderRlp The full header of the closed OSTblock,
+     *                                 RLP encoded.
+     *
+     * @return The full header of the new, opened OSTblock as a result of the
+     *         closing, RLP encoded. The newly opened OSTblock header inculdes
+     *         all changes in the set of validators.
+     */
+    function closeOstBlock(
+        bytes _closedOstBlockHeaderRlp
+    )
+        external
+        returns (bytes _openedOstBlockHeaderRlp);
+
+    /**
+     * @notice Returns the total stake of all validators at a given height.
+     *
+     * @param _height The height for which to get the total deposit.
+     *
+     * @return The total stake at the given height.
+     */
+    function totalStakeAtHeight(
+        uint256 _height
+    )
+        external
+        view
+        returns (uint256 totalStake_);
+
+    /**
+     * @notice Returns the stake of a validator at a specific OSTblock height,
+     *         based on the auxiliary address of the validator.
+     *
+     * @dev The OriginCore can use this method to track the verified stake by
+     *      the verified votes and notice when a supermajority has been
+     *      reached, therefore committing the OSTblock.
+     *
+     * @param _height The OSTblock height for which to get the stake.
+     * @param _auxiliaryValidatorAddress The address of the validator on the
+     *                                   auxiliary chain.
+     *
+     * @return The stake of the validator at the given height. Can be 0, for
+     *         example when the address never deposited.
+     */
+    function getStakeAtHeightByAuxiliaryAddress(
```

Comment:
`weight(uint256 _height, address _validator)`, only for open height, use height to assert